AppDashboard/lib/app_dashboard_data.py

```diff
@@ -351,91 +284,43 @@ def update_head_node_ip(self):
-        in this AppScale deployment.
-    """
-    servers = self.get_all(ServerStatus)
-    return [{'ip': server.key.id(), 'cpu': server.cpu,
-             'memory': server.memory, 'disk': server.disk,
-             'roles': server.roles,
-             'key': server.key.id().translate(None, '.')} for server in servers]
-
-  def update_status_info(self):
-    """ Queries the AppController to get status information for all servers in
-    this deployment, storing it in the Datastore for later viewing.
+    Args:
+      app_id: A string, the application identifier.
     """
     try:
-      nodes = self.helper.get_appcontroller_client().get_cluster_stats()
-      updated_statuses = []
-      for node in nodes:
-        status = self.get_by_id(ServerStatus, node['public_ip'])
-        cpu_usage = 100.0 - node['cpu']['idle']
-        memory_usage = round(100.0 * node['memory']['used'] /
-                             node['memory']['total'], 1)
-        total_disk = 0
-        total_used = 0
-        #TODO: instead of totals display disk usage per disk?
-        for disk in node['disk']:
-          for _, disk_info in disk.iteritems():
-            total_disk += disk_info['total']
-            total_used += disk_info['used']
-        disk_usage = round(100.0 * total_used / total_disk, 1)
-        if status:
-          # Make sure that at least one field changed before we decide to
-          # update this ServerStatus.
-          if status.cpu != str(cpu_usage) or \
-                  status.memory != str(memory_usage) or \
-                  status.disk != str(disk_usage) or \
-                  status.roles != node['roles']:
-            status.cpu = str(cpu_usage)
-            status.memory = str(memory_usage)
-            status.disk = str(disk_usage)
-            status.roles = node['roles']
-            updated_statuses.append(status)
-        else:
-          status = ServerStatus(id=node['public_ip'])
-          status.cpu = str(cpu_usage)
-          status.memory = str(memory_usage)
-          status.disk = str(disk_usage)
-          status.roles = node['roles']
-          updated_statuses.append(status)
-      ndb.put_multi(updated_statuses)
+      request_info = self.helper.get_appcontroller_client()\
+                                  .get_request_info(app_id)
+      timestamp = datetime.datetime.fromtimestamp(request_info.get('timestamp'))
+      lastHourDateTime = timestamp - datetime.timedelta(hours=1)
+      old_requests_query = RequestInfo.query(RequestInfo.timestamp <
+                                             lastHourDateTime)
+      old_requests = []
+      for key in old_requests_query.iter(keys_only=True):
+        old_requests.append(key)
+      ndb.delete_multi(old_requests)
+      reversed_time = (2 ** 34 - int(request_info.get('timestamp'))) * 1000000
+      request_stats = RequestInfo(
+                      id=app_id + str(reversed_time),  # puts entities time desc.
```

Comment:
@cdonati ok I'll change that, that was how it was previously done so I had kept it.